index.js

```diff
@@ -121,8 +124,19 @@ function magnetURIEncode (obj) {
 
   // support using convenience names, in addition to spec names
   // (example: `infoHash` for `xt`, `name` for `dn`)
-  if (obj.infoHashBuffer) obj.xt = `urn:btih:${obj.infoHashBuffer.toString('hex')}`
-  if (obj.infoHash) obj.xt = `urn:btih:${obj.infoHash}`
+
+  // Deduplicate xt by using a set
+  let xts = new Set()
+  if (obj.xt && typeof obj.xt === 'string') xts.add(obj.xt)
+  if (obj.xt && typeof obj.xt === 'object') xts = new Set(obj.xt)
+  if (obj.infoHashBuffer) xts.add(`urn:btih:${obj.infoHashBuffer.toString('hex')}`)
+  if (obj.infoHash) xts.add(`urn:btih:${obj.infoHash}`)
+  if (obj.infoHashV2Buffer) xts.add(obj.xt = `urn:btmh:1220${obj.infoHashV2Buffer.toString('hex')}`)
+  if (obj.infoHashV2) xts.add(`urn:btmh:1220${obj.infoHashV2}`)
+  const xtsDeduped = Array.from(xts)
+  if (xtsDeduped.length === 1) obj.xt = xtsDeduped[0]
+  if (xtsDeduped.length > 1) obj.xt = xtsDeduped
```

Comment:
this is my only concerning line, can `obj.xt` be an array? (I'm super outdated in the library), will check later.

Reply:
[I think so.](https://github.com/webtorrent/magnet-uri#encode) Also, this behavior is tested by two tests.